presto-native-execution/presto_cpp/main/connectors/PrestoToVeloxConnector.cpp

```diff
@@ -846,33 +1011,97 @@ std::unique_ptr<connector::ConnectorTableHandle> toHiveTableHandle(
+        names.emplace_back(std::move(name));
+        if (auto sortedRangeSet =
+                std::dynamic_pointer_cast<protocol::SortedRangeSet>(
+                    domain.second.values)) {
+          auto type = stringToType(sortedRangeSet->type, typeParser);
+          types.push_back(VELOX_DYNAMIC_TYPE_DISPATCH(
+              fieldNamesToLowerCase, type->kind(), type));
+        } else {
+          VELOX_UNSUPPORTED("Unsupported filter found.");
+        }
+      }
+    }
+#endif
     finalDataColumns = ROW(std::move(names), std::move(types));
   }
 
   if (tableParameters.empty()) {
-    return std::make_unique<connector::hive::HiveTableHandle>(
-        tableHandle.connectorId,
-        tableName,
-        isPushdownFilterEnabled,
-        std::move(subfieldFilters),
-        remainingFilter,
-        finalDataColumns);
+    if (canUseCudfTableScan()) {
+#ifdef PRESTO_ENABLE_CUDF
+      facebook::velox::core::TypedExprPtr filterExpr = nullptr;
+      if (!subfieldFilters.empty()) {
+        filterExpr = getTypedExprFromSubfieldFilter(
+            domains, exprConverter, typeParser, finalDataColumns);
+      }
+      return std::make_unique<
+          cudf_velox::connector::parquet::ParquetTableHandle>(
+          "parquet-test",
+          tableName,
+          isPushdownFilterEnabled,
+          filterExpr,
+          remainingFilter,
+          finalDataColumns);
+#else
+      // this never happens because if PRESTO_ENABLE_CUDF
+      // is false then canUseCudfTableScan returns false
+      // and we never hit this branch
+      return nullptr;
+#endif
+    } else {
+      return std::make_unique<connector::hive::HiveTableHandle>(
+          tableHandle.connectorId,
+          tableName,
+          isPushdownFilterEnabled,
+          std::move(subfieldFilters),
+          remainingFilter,
+          finalDataColumns);
+    }
   }
 
   std::unordered_map<std::string, std::string> finalTableParameters = {};
   finalTableParameters.reserve(tableParameters.size());
   for (const auto& [key, value] : tableParameters) {
     finalTableParameters[key] = value;
   }
-
-  return std::make_unique<connector::hive::HiveTableHandle>(
-      tableHandle.connectorId,
-      tableName,
-      isPushdownFilterEnabled,
-      std::move(subfieldFilters),
-      remainingFilter,
-      finalDataColumns,
-      finalTableParameters);
+  if (canUseCudfTableScan()) {
+#ifdef PRESTO_ENABLE_CUDF
+    facebook::velox::core::TypedExprPtr filterExpr = nullptr;
+    if (!subfieldFilters.empty()) {
+      filterExpr = getTypedExprFromSubfieldFilter(
+          domains, exprConverter, typeParser, finalDataColumns);
+    }
+    return std::make_unique<cudf_velox::connector::parquet::ParquetTableHandle>(
+        "parquet-test",
+        tableName,
+        isPushdownFilterEnabled,
+        filterExpr,
+        remainingFilter,
+        finalDataColumns);
+#else
+    // this never happens - if PRESTO_ENABLE_CUDF is not defined,
```

Comment:
I think you could add a `VELOX_FAIL/UNREACHABLE` (I forgot the macro's name) here instead of returning a nullptr